src/System.Management.Automation/engine/CommandCompletion/CompletionAnalysis.cs

```diff
@@ -503,6 +503,11 @@ internal List<CompletionResult> GetResultHelper(CompletionContext completionCont
                                 return completions;
                             }
                         }
+                        else if (lastAst.Parent is IndexExpressionAst indexExpressionAst)
+                        {
+                            completionContext.WordToComplete = (tokenAtCursor as StringToken).Value;
```

Comment:
Can it be null?

Reply:
No, the switch ensures it's one of:
```
TokenKind.StringExpandable
TokenKind.StringLiteral
```
and even an empty or incomplete string token will have an empty string as the value.

Reply:
Sorry, I mean `tokenAtCursor as StringToken`.

Reply:
Yes, I meant the whole statement can never be null. `tokenAtCursor` can never be null because the switch statement uses the `tokenAtCursor` as input. `StringToken` is always a valid conversion because the switch ensures it's one of those 2 token kinds which can always be converted to a stringtoken with at the very least an empty string value.